playbook-website/app/controllers/pages_controller.rb

```diff
@@ -26,6 +26,53 @@ def enable_dark_mode
     redirect_back(fallback_location: root_path)
   end
 
+  def to_url_format(text)
+    text.gsub(/[^a-zA-Z0-9]+/, "-").strip.gsub(/\s+/, "-")
+  end
+
+  def extract_changelog_data(changelog)
+    releases = []
+
+    changelog.split(/^# /).each do |section|
+      break if releases.size == 2
+
+      next unless section.strip.length.positive?
+
+      title_match = section.match(/(.+?)\n/)
+      next unless title_match
```

Comment:
these `next unless` are good to be sure the UI doesn't break, but we might want to consider adding some kind of linting rules to the `CHANGELOG.md` so that we require certain info/formats (out of scope for this story)